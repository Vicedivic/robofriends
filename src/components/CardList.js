import React from "react";
import Card from "./Card";

const CardList = ({ robots }) => {
  return (
    <div>
      {robots.map(robot => (
        <Card {...robot} key={robot.id} />
      ))}
    </div>
  );
};

export default CardList;
