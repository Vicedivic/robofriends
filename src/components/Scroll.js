import React from "react";

const Scroll = ({ children }) => {
  return (
    <div style={{ overflowY: "scroll", border: "1px solid black", height: 500 }}>
      {children}
    </div>
  );
};

export default Scroll;
