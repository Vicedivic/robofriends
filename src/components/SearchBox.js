import React from "react";

const SearchBox = ({ searchChange }) => {
  return (
    <div>
      <label htmlFor="search" style={{ fontSize: 0, position: "absolute" }}>Search</label>
      <input
        className="search"
        type="search"
        placeholder="search robots"
        onChange={searchChange}
        id="search"
      />
    </div>
  );
};

export default SearchBox;
