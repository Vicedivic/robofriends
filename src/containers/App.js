import React, { useEffect } from "react";
import { connect } from "react-redux";
import CardList from "../components/CardList";
import SearchBox from "../components/SearchBox";
import Scroll from "../components/Scroll";
import ErrorBoundry from "../components/ErrorBoundry";
import { setSearchField } from "../redux/actions/search";
import { getRobots } from "../redux/actions/robot";

export const App = ({ searchField, setSearchField, robot, getRobots }) => {
  const { robots, pending, error } = robot;

  useEffect(() => {
    getRobots();
  }, []);

  const filteredRobots = robots.filter(robots => {
    return robots.name.toLowerCase().includes(searchField.toLowerCase());
  });

  return pending ? (
    <h1>Loading</h1>
  ) : (
    <div className="tc">
      <h1>RoboFriends</h1>
      <SearchBox searchChange={e => setSearchField(e.target.value)} />
      <Scroll>
        <ErrorBoundry>
          <CardList robots={filteredRobots} />
        </ErrorBoundry>
      </Scroll>
    </div>
  );
};

const mapStateToProps = state => ({
  searchField: state.search.searchField,
  robot: state.robot
});

export default connect(mapStateToProps, { setSearchField, getRobots })(App);
