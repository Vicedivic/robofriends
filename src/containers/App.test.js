import React from "react";
import { create, act } from "react-test-renderer";
import { mount } from "enzyme";
import ShallowRenderer from "react-shallow-renderer";
import { App } from "./App";
import robotData from "../robots";
import SearchBox from "../components/SearchBox";
import Scroll from "../components/Scroll";
import CardList from "../components/CardList";

describe("App", () => {
  const mockGetRobots = jest.fn();
  const props = {
    searchField: "",
    setSearchField: jest.fn(),
    robot: { robots: robotData, pending: false, error: null },
    getRobots: mockGetRobots,
  };
  const mockOnSearchChange = jest.fn();
  const renderer = new ShallowRenderer();
  const appHtml = renderer.render(<App {...props} />);
  const app = mount(
    <App
      {...props}
      children={
        <div className="tc">
          <h1>RoboFriends</h1>
          <SearchBox searchChange={mockOnSearchChange} />
          <Scroll>
            <CardList robots={robotData} />
          </Scroll>
        </div>
      }
    />
  );

  it("renders properly", () => {
    expect(appHtml).toMatchSnapshot();
  });

  it("calls the get robots action on mount", () => {
    act(() => {
      create(<App {...props} />);
    });
    expect(mockGetRobots).toHaveBeenCalled();
  });

  describe("when input is typed into the search field", () => {
    beforeEach(() => {
      app
        .find("SearchBox")
        .find("input")
        .simulate("change", { target: { value: "test" } });
    });

    it("calls the on search function", () => {
      // expect(mockOnSearchChange.mock.calls[0][0]).toBe("test");
    });
  });
});
