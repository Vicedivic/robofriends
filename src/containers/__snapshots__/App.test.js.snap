// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<div
  className="tc"
>
  <h1>
    RoboFriends
  </h1>
  <SearchBox
    searchChange={[Function]}
  />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={
          Array [
            Object {
              "email": "sincere@april.biz",
              "id": 1,
              "name": "Leanne Graham",
            },
            Object {
              "email": "shanna@melissa.tv",
              "id": 2,
              "name": "Ervin Howell",
            },
            Object {
              "email": "nathan@yesenia.net",
              "id": 3,
              "name": "Clementine Bauch",
            },
            Object {
              "email": "julianna.oconner@kory.org",
              "id": 4,
              "name": "Patricia Lebsack",
            },
          ]
        }
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
